src/foam/u2/FormattedTextField.js

```diff
@@ -106,14 +111,16 @@ foam.CLASS({
       this.isDelete = false;
       this.includeTrailingDelimiter = true;
       this.formatted = false;
+      if ( this.hasOwnProperty('formattedData') )
+        this.data = this.returnFormatted ? this.formattedData : this.formattedData.replace(/\D/g,'');
     }
   ],
 
   listeners: [
     {
       name: 'formatData',
       code: function () {
-        this.data = this.formattedData.replace(/\D/g,'');
+        var data = this.formattedData.replace(/\D/g,'');
```

Comment:
You check for hasOwnProperty formattedData above, but use it here without checking.  Suspicious.

Reply:
the check is only needed when resetState is called from the first line of inite, because at that point formattedData is null. but this method is only called whenever formattedData is set, so it shouldn't be neededd